src/player.js

```diff
@@ -657,10 +656,9 @@ export default class Player extends FakeEventTarget {
       statsObject.targetBuffer = this._engine.targetBuffer;
       statsObject.availableBuffer = this._engine.availableBuffer;
       // consideration of the end of the playback in the target buffer calc
-      if (!this.isLive()) {
-        statsObject.targetBuffer = videoElement
-          ? Math.min(statsObject.targetBuffer, videoElement.duration - videoElement.currentTime)
-          : statsObject.targetBuffer;
+      if (!this.isLive() && this.getVideoElement()) {
+        const videoElement = this.getVideoElement();
+        statsObject.targetBuffer = Math.min(statsObject.targetBuffer, videoElement.duration - videoElement.currentTime);
```

Comment:
ok, I now understand this is not safe, as accessing the videoElement duration and currentTime is not guaranteed, as not all are HTML5 video element.
So I guess best place at the moment is in the base media source adapter in the `targetBuffer` getter